index.js

```diff
@@ -5,8 +5,31 @@
  * MIT Licensed
  */
 var promiseExists = typeof Promise === 'function';
-var globalObject = typeof window !== 'undefined' ? window : typeof global !== 'undefined' ? global : self; // eslint-disable-line
-var isDom = 'location' in globalObject && 'document' in globalObject;
+
+/* eslint-disable */
+// See http://stackoverflow.com/a/6930376
+var globalObject;
+try {
+  globalObject = Function('return this')() || (42, eval)('this');
```

Comment:
On top of what @meeber has mentioned, I would prefer `new Function`. `new` with constructors is the new black in ES6. Also, this won't work in Chrome OS extensions for example. More on CSP: https://github.com/jashkenas/underscore/issues/906.